receiver/sqlqueryreceiver/db_client.go

```diff
@@ -0,0 +1,112 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlqueryreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlqueryreceiver"
+
+import (
+	"context"
+	"database/sql"
+	"strconv"
+	"time"
+
+	// register db drivers
+	_ "github.com/SAP/go-hdb/driver"
+	_ "github.com/denisenkom/go-mssqldb"
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/lib/pq"
+	_ "github.com/snowflakedb/gosnowflake"
+	"go.uber.org/zap"
+)
+
+type dbClient interface {
+	metricRows(ctx context.Context) ([]metricRow, error)
+}
+
+type dbSQLClient struct {
+	db     *sql.DB
+	logger *zap.Logger
+	sql    string
+}
+
+func newDbClient(db *sql.DB, sql string, logger *zap.Logger) dbClient {
+	return dbSQLClient{
+		db:     db,
+		sql:    sql,
+		logger: logger,
+	}
+}
+
+type metricRow map[string]string
+
+func (cl dbSQLClient) metricRows(ctx context.Context) ([]metricRow, error) {
+	sqlRows, err := cl.db.QueryContext(ctx, cl.sql)
+	if err != nil {
+		return nil, err
+	}
+	var out []metricRow
+	rr := reusableRow{
+		attrs: map[string]func() string{},
+	}
+	types, err := sqlRows.ColumnTypes()
+	if err != nil {
+		return nil, err
+	}
+	for _, columnType := range types {
+		colName := columnType.Name()
+		switch columnType.ScanType().Name() {
```

Comment:
Mind linking to https://github.com/golang/go/blob/29057b707d468bd690e6eadfebe89c4ff737867c/src/database/sql/sql.go#L3210 in a comment?

Don't think it's necessary in these changes and I'm unsure of their prevalence but we could handle most of the uint cases. I wonder if floats should go in now.